### PR TITLE
Organize Balancer Spells into folders

### PR DESCRIPTION
**Pull request by viniabussafi**
With the upcoming V3  basically doubling the number of spells for Balancer, this PR aims to organize our folders into sectors, making blockchain-specific folders easier to navigate. 
There are no changes to the spells themselves, except for removing legacy trade spells. 
All schema files have been renamed with "_schema" and improved for syntax, formatting, and column descriptions. 
Outputs have been validated and remain unchanged, as expected.

### CODE REVIEW

**Comment by viniabussafi** [comment]
Hey, @Hosuke!
Could you please give me a hand with setting this up, if possible? 
I explained the idea on this [comment](https://github.com/duneanalytics/spellbook/pull/5958#issuecomment-2124530349).
Will this work? If so, how should I setup the dbt_project.yml file?
thank you!

**Comment by Hosuke** [comment]
I see, I will remove balancer from `dbt_project.yml` first.
Please ensure `schema` existed in all config blocks.

**Comment by viniabussafi** [comment]
Tests pass now and PR is good to go on my end. 
I know it may take a while to review, so thanks for your work :) !

File: models/balancer/balances/balancer_token_balance_changes.sql [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{{ config(
+    schema = 'balancer',
+    alias = 'token_balance_changes',
+    post_hook='{{ expose_spells(\'["arbitrum", "avalanche_c", "base", "ethereum", "gnosis", "optimism", "polygon", "zkevm"]\',
+                                "project",
+                                "balancer",
+                                \'["viniabussafi"]\') }}'
```
Comment (Hosuke) [inline models/balancer/balances/balancer_token_balance_changes.sql:7]: We may apply the new format for all post_hook in this PR:
```suggestion
    post_hook='{{ expose_spells(blockchains = \'["arbitrum", "avalanche_c", "base", "ethereum", "gnosis", "optimism", "polygon", "zkevm"]\',
                                spell_type = "project",
                                spell_name = "balancer",
                                contributors = \'["viniabussafi"]\') }}'
```

**Review by Hosuke — CHANGES_REQUESTED** [review]

File: models/balancer/balances/balancer_token_balance_changes.sql (hunk 1 above)
Comment (viniabussafi) [reply models/balancer/balances/balancer_token_balance_changes.sql:7]: will work on it!

Comment (viniabussafi) [reply models/balancer/balances/balancer_token_balance_changes.sql]: done! also took the chance to remove some post hooks from chain-specific spells in order to clean up the explorer

**Comment by jeff-dude** [comment]
love to the initiative to organize your growing number of spells.

at first glance, my feedback:
- we should take advantage of the `models/_project/` directory and put `balancer` directly in there (looks like already done [here](https://github.com/duneanalytics/spellbook/tree/main/models/_project/balancer))
- next level in that directory should likely be sector/spell type/whatever makes sense here (i.e. balances, bpt, pools, etc)
- then each chain folder within each of the above
- then files within that chain directory

something like:
`models/_project/balancer/bpt/arbitrum/<spells/schemas here>`

what do ya think?

**Comment by viniabussafi** [comment]
Makes perfect sense, @jeff-dude! I'm on it 🫡

**Review by jeff-dude — APPROVED** [review]
works for me!

**Comment by jeff-dude** [comment]
> There are no changes to the spells themselves

based on this comment, i will merge without full refresh enabled in prod to avoid fully rebuilding it all

**Comment by jeff-dude** [comment]
okay looks like poor assumptions were made on my part. this PR did require `state:modified` to be enabled in order to build / rebuild some spells. prod is now failing on missing dependencies. i'm not sure if this will work, but to make it easy as possible, i will try to revert this PR, revert the revert and hope prod picks this whole thing up as modified and runs it all. otherwise, i'll have to go model by model and run and that introduces risk of missing some

**Pull request merged**